### ISSUE TITLE
Error on run/verify when server starts slowly

### ISSUE THREAD

**Issue by shuhei** [workshopper/learnyoureact#22]
When I run/verify program.js and my server starts slowly, I get the following error.

```
✗ Error connecting to http://localhost:14880: ECONNREFUSED
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: connect ECONNREFUSED
    at exports._errnoException (util.js:746:11)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:983:19)
```

It seems that [exercise programs wait only 1 second](https://github.com/tako-black/learnyoureact/blob/master/exercises/isomorphic/exercise.js#L80). It can be problematic on low-spec machines.

**Comment by kohei-takata** [comment]
@shuhei 
Fixed!

**Comment by shuhei** [comment]
Great! Thanks!

**Comment by gustavjf** [comment]
Hi @tako-black, I am still having this issue when running `learnyoureact verify program`.

Node: 0.12.7
learnyoureact: 0.9.10

At first it happened only once after I waited a bit of time. Now it's happening every time I try to run verify. After a long time, it works again so it's a bit unpredictable. Let me know if you need any more info.

**Comment by kohei-takata** [comment]
Hi @gustavjf, thanks for your comment!
This error happens when the server of `learnyoureact` waits 3 seconds.
Could you restart your machine or shut down some heavy program of your machine?

**Comment by scottnath** [comment]
@tako-black I am having this issue as well

Node: v5.0.0
learnyoureact@0.10.1

It looks like learnyoureact is trying to reach a second port, is that what's supposed to happen? I'm getting this error with hello and components.

```
✗ Error connecting to http://localhost:49351: ECONNREFUSED
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: connect ECONNREFUSED 127.0.0.1:49351
    at Object.exports._errnoException (util.js:860:11)
    at exports._exceptionWithHostPort (util.js:883:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
```

I put some console.log's in the code to see what was happening, and it looks like it fails when trying to call the second port. For instance, I added logging at line [77 of component's exercise.js](https://github.com/tako-black/learnyoureact/blob/master/exercises/components/exercise.js#L77):

```
    console.log('url: '+url)
        function error (err) {
            exercise.emit('fail', 'Error connecting to http://localhost:' + port + ': ' + err.code)
        }
```

and got the following two responses before receiving the `ECONNREFUSED` error.
url1: http://localhost:49350
url2: http://localhost:49351

From the error above, it's happening on the second port `49351`

note: I also tried making the timeout 10000, but that didn't help. I do not have any other heavy processes running which would interfere with this script.

note 2: running `learnyoureact run program.js` returns the correct response, with no errors

**Comment by scottnath** [comment]
@tako-black most likely unrelated, but in my testing I used the latest babel to see if the older version was causing issues. I found that this line:

```
require('babel/register');
```

had to be rewritten like so:

```
require('babel-core/register');
```

thanks,
Scott

**Comment by kohei-takata** [comment]
@scottnath 
Thanks for your reaport.
I've open a new issue https://github.com/tako-black/learnyoureact/issues/72.
I will fix at the issue.
